csi/controller.go

```diff
@@ -371,6 +392,63 @@ func (s *OsdCsiServer) DeleteVolume(
 	return &csi.DeleteVolumeResponse{}, nil
 }
 
+// ControllerExpandVolume is a CSI API which resizes a volume
+func (s *OsdCsiServer) ControllerExpandVolume(
+	ctx context.Context,
+	req *csi.ControllerExpandVolumeRequest,
+) (*csi.ControllerExpandVolumeResponse, error) {
+	if len(req.GetVolumeId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume id must be provided")
+	} else if req.GetCapacityRange() == nil {
+		return nil, status.Error(codes.InvalidArgument, "Capacity range must be provided")
+	} else if req.GetCapacityRange().GetRequiredBytes() < 0 || req.GetCapacityRange().GetLimitBytes() < 0 {
+		return nil, status.Error(codes.InvalidArgument, "Capacity ranges values cannot be negative")
+	}
+
+	// Get parameters
+	spec := &api.VolumeSpecUpdate{}
+
+	// Get Size
+	newSize := uint64(req.GetCapacityRange().GetRequiredBytes())
+	if newSize != 0 {
```

Comment:
What happens if it is `0`? The code continues, without anything to update. Should it instead check for zero in line 404?

Reply:
This was a typo I think. We don't need to check for 0.

Since RequiredBytes and LimitBytes are both optional per the spec, if you don't pass one, it'll use the zero value.